src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -453,7 +453,14 @@ private void LogTranscriptHeader(System.Management.Automation.Remoting.PSSenderI
                     psVersionInfo.AppendLine(versionKey + ": " + valueString);
                 }
             }
-
+            string psConfigurationName = null;
+            if (senderInfo != null && !string.IsNullOrEmpty(senderInfo.ConfigurationName))
+            {
+                string[] configurationArray =
+                    senderInfo.ConfigurationName.Split(new char[] {System.IO.Path.DirectorySeparatorChar},
+                        StringSplitOptions.RemoveEmptyEntries);
+                psConfigurationName = configurationArray[configurationArray.Length - 1];
```

Comment:
This may be more efficient for extracting the configuration name from the Shell Uri since it only allocates one string instead of multiple string objects:

```
string shellPrefix = System.Management.Automation.Remoting.Client.WSManNativeApi.ResourceURIPrefix;
int index = shelluri.IndexOf(shellPrefix, StringComparison.OrdinalIgnoreCase);
return (index == 0) ? shelluri.Substring(shellPrefix.Length) : string.Empty;
```